ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java

```diff
@@ -123,6 +137,14 @@ public long writeTo(WritableByteChannel channel) throws IOException {
         bufferToSend.putInt(usermetadata.capacity());
         bufferToSend.put(usermetadata);
         bufferToSend.putShort((short) blobType.ordinal());
+        // @todo remove this check once V4 is everywhere.
+        if (getVersionId() == PUT_REQUEST_VERSION_V4) {
+          short keyLength = blobKey == null ? 0 : (short) blobKey.length;
```

Comment:
functionality question: is this allowed to be none? After EAR is implemented, there will still be a situation where something is not encrypted?

Reply:
yes, as per our current design, account owners get to enable/disable encryption at container level. @nsivabalan please chime in.